src/test/kotlin/no/nav/helse/inntektsmelding/db/PostgresTest.kt

```diff
@@ -0,0 +1,51 @@
+package no.nav.helse.inntektsmelding.db
+
+import com.zaxxer.hikari.HikariDataSource
+import org.junit.Test
+import org.testcontainers.containers.PostgreSQLContainer
+import kotlin.test.assertEquals
+
+internal class PostgresTest {
+
+    @Test
+    fun `Migration scripts are applied successfully`() {
+        withCleanDb {
+            val migrations = migrate(DataSource.instance)
+            assertEquals(1, migrations, "Wrong number of migrations")
+        }
+    }
+
+    @Test
+    fun `Migration scripts are idempotent`() {
+        withCleanDb {
+            migrate(DataSource.instance)
+
+            val migrations = migrate(DataSource.instance)
+            assertEquals(0, migrations, "Wrong number of migrations")
```

Comment:
Tester dette egentlig at db-scriptene er idempotente? For grunnen til at ingen migreringer blir kjørt er vel at Flyway ser at alle scriptene allerede er kjørt ?   Og for å teste at de var idempotente så måtte de vel egentlig også kjørt andre gang ? :)

Reply:
Bare å innrømme først som sist at jeg har kopiert testen fra https://github.com/navikt/dp-inntekt-api 😅 Enig - dette virker litt snålt.

Reply:
Det blir riktig, faktisk - migrate blir kalt to ganger, og så sjekker man at det ikke har skjedd noen endringer ved andre kjøring

Reply:
Nei. Du tester at Flyway.migrate()-metoden er idempotent på samme script-sett (at Flyway-rammeverket gjør det den sier den skal gjøre m.a.o.), altså at at når du kaller .migrate() en andre gang på samme script-sett, så kjører den ikke scriptene på nytt (i.e: 0 migreringert kjørt).
Denne testen vil bli grønn selv om db-scriptet ikke hadde vært idempotent (altså, selv om det ikke hadde hatt "IF NOT EXISTS" på create table), vil den ikke ?

Reply:
Dette er en litt sånn "kjekt å ha"-test for meg - hva mener du skulle vært annerledes?

Reply:
Skjønner. Holder å endre navnet på testen for at jeg blir fornøyd ;)   Vet det finnes en del indidivider i "migrerings-mijøet" som mener det er best practice at hvert enkelt db-script er idempotent selv om man bruker Flyway. I den konteksten er det derfor fort gjort å tro at denne testen tester nettopp dette, p.g.a. navnet, selv om den ikke gjør det. Burde kanskje hett noe som "Migration scripts are run only once" i stedet.